packages/fuels/test/features/dev.integration.test.ts

```diff
@@ -0,0 +1,85 @@
+import { execSync, spawn } from 'child_process';
+import { mkdirSync, rmSync } from 'fs';
+import { tmpdir } from 'os';
+import path from 'path';
+
+import { randomUUID, sleep } from '../../src';
+import { findChildProcessPid } from '../utils/findChildProcessPid';
+import { isProcessRunning } from '../utils/isProcessRunning';
+
+function runInit() {
+  const fuelsPath = path.join(process.cwd(), 'packages/fuels');
+
+  const init = path.join(tmpdir(), '.fuels', 'tests', 'dev', randomUUID());
+  mkdirSync(init, { recursive: true });
+
+  execSync(`pnpm init && pnpm link ${fuelsPath}`, { cwd: init });
+  const contractDir = path.join(init, 'contract');
+  const outputDir = path.join(init, 'output');
+  mkdirSync(contractDir);
+  mkdirSync(outputDir);
+  execSync(`${process.env.FORC_PATH} init`, { cwd: contractDir });
+  execSync(`pnpm fuels init -o ${outputDir} -c ${contractDir} --path ${init} --fuel-core-port 0`);
+
+  return {
+    init,
+    [Symbol.dispose]: () => {
+      rmSync(init, { recursive: true });
+    },
+  };
+}
+
+/**
+ * @group node
+ */
+describe('dev2', () => {
+  it(
+    'cleans up resources on graceful shutdown',
+    async () => {
+      using paths = runInit();
```

Comment:
The `using` keyword is great because you don't have to think about where the test might fail so that you put the `cleanup` if it does fail there. With `using`, the cleanup will happen automatically both when the test finishes as well as when it throws an error.